695_Max_Area_of_Island.py

```diff
@@ -0,0 +1,33 @@
+# n = len(grid)
+# m = len(grid[0])
+# time: O(n*m)
+# space: O(1)
```

Comment:
The space complexity of this approach is not constant because it uses recursive calls. When programs call a function recursively, it creates a new stack frame, and it uses some memory. You have to consider this memory consumption.